app/src/main/java/com/example/mobile_development/components/login.kt

```diff
@@ -0,0 +1,159 @@
+import androidx.compose.material.icons.filled.Visibility
+import androidx.compose.material.icons.filled.VisibilityOff
+import androidx.compose.material3.Button
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.example.mobile_development.R
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun LoginScreen(navigateToHomeScreen: () -> Unit = {}, modifier: Modifier = Modifier) {
+    var (email, setEmail) = remember { mutableStateOf("") }
+    var (password, setPassword) = remember { mutableStateOf("") }
+    var (showPass, setShowPass) = remember { mutableStateOf(false) }
+    var (isError, setIsError) = remember { mutableStateOf(false) }
+
+    fun validateEmail(): Boolean {
+        if (!TextUtils.isEmpty(email)) {
+            var isValid =
+                email.length >= 3 && android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()
+            setIsError(isValid)
+            return isValid
+        }
+        return true
+    }
+
+    fun validatePassword(): Boolean {
+        if (!TextUtils.isEmpty(password)) {
+            var isValid = password.length >= 3 && password.length <= 30
+            setIsError(isValid)
+            return isValid
+        }
+        return true
+    }
+
+    Image(
+        modifier = Modifier.fillMaxSize(),
+        painter = painterResource(R.drawable.etherealartefacts_login_background),
+        contentDescription = "background_image",
+        contentScale = ContentScale.FillBounds
+    )
+    Column(
+        modifier = modifier
+            .padding(horizontal = 15.dp, vertical = 35.dp)
+            .fillMaxSize()
+    ) {
+        Image(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(vertical = 40.dp),
+            imageVector = ImageVector.vectorResource(id = R.drawable.ethereal_artefacts_logo),
+            contentDescription = "logo-image",
+            contentScale = ContentScale.Inside
+        )
+
+        Text(
+            text = "Log in",
+            modifier = modifier,
```

Comment:
Here you might also supply external modifier values by accident. If the composable doesn't need a modifier, you can just opt out of it and have none.